### PR TITLE
feat:  multisig: lotus-sheed miner-multisig change-worker command.

### PR DESCRIPTION
**Pull request by spark8899**
## Related Issues
[lotus-shed miner-multisig add change-worker command](https://github.com/filecoin-project/lotus/issues/8280)

## Proposed Changes
Added two subcommands, propose-change-worker and Confirm-change-worker.

## Additional Info
lotus-shed miner-multisig --from <sigle-address> --miner <miner-actor> --multisig <multisigAddress messageId> propose-change-worker <new-worker>
lotus msig approve --from <approve-account> <multisigAddress messageId> <proposerAddress destination value>

lotus-shed miner-multisig --from <sigle-address> --miner <miner-actor> --multisig <multisigAddress messageId> Confirm-change-worker <new-worker>
lotus msig approve --from <approve-account> <multisigAddress messageId> <proposerAddress destination value>

## Checklist

Before you mark the PR ready for review, please make sure that:
- [x] All commits have a clear commit message.
- [x] The PR title is in the form of of `<PR type>: <area>: <change being made>`
    - example: ` fix: mempool: Introduce a cache for valid signatures`
    - `PR type`: _fix_, _feat_, _INTERFACE BREAKING CHANGE_, _CONSENSUS BREAKING_, _build_, _chore_, _ci_, _docs_,_perf_, _refactor_, _revert_, _style_, _test_
    - `area`: _api_, _chain_, _state_, _vm_, _data transfer_, _market_, _mempool_, _message_, _block production_, _multisig_, _networking_, _paychan_, _proving_, _sealing_, _wallet_, _deps_
- [ ] This PR has tests for new functionality or change in behaviour
- [ ] If new user-facing features are introduced, clear usage guidelines and / or documentation updates should be included in https://lotus.filecoin.io or [Discussion Tutorials.](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] CI is green

### CODE REVIEW

File: cmd/lotus-shed/miner-multisig.go [hunk 1]
```diff
@@ -8,6 +8,8 @@ import (
 	"github.com/filecoin-project/go-state-types/abi"
 	miner5 "github.com/filecoin-project/specs-actors/v5/actors/builtin/miner"
 
+    miner2 "github.com/filecoin-project/specs-actors/v2/actors/builtin/miner"
```
Comment (magik6k) [inline cmd/lotus-shed/miner-multisig.go:11]: gofmt complains about this:
```suggestion
	miner2 "github.com/filecoin-project/specs-actors/v2/actors/builtin/miner"
```

File: cmd/lotus-shed/miner-multisig.go [hunk 2]
```diff
@@ -368,6 +373,301 @@ var mmApproveChangeOwner = &cli.Command{
...
+			fmt.Printf("Exit Code: %d\n", retval.Code)
+			fmt.Printf("Return Value: %x\n", retval.Ret)
+		}
+
+		return nil
+	},
+}
+
+var mmConfirmChangeWorker = &cli.Command{
+	Name:      "Confirm-change-worker",
```
Comment (magik6k) [inline cmd/lotus-shed/miner-multisig.go:474]:
```suggestion
	Name:      "confirm-change-worker",
```

File: cmd/lotus-shed/miner-multisig.go [hunk 3]
```diff
@@ -368,6 +373,301 @@ var mmApproveChangeOwner = &cli.Command{
...
+			fmt.Printf("Transaction was executed during propose\n")
+			fmt.Printf("Exit Code: %d\n", retval.Code)
+			fmt.Printf("Return Value: %x\n", retval.Ret)
+		}
+		return nil
+	},
+}
+
+var mmProposeControlSet = &cli.Command{
+	Name:      "ProposeControlSet",
```
Comment (magik6k) [inline cmd/lotus-shed/miner-multisig.go:557]:
```suggestion
	Name:      "propose-control-set",
```

**Review by magik6k — APPROVED** [review]
Msig worker address won't really work (it's impossible to sign blocks with msig worker address), but I guess utils to undo setting an msig worker are nice to have.

**Comment by silvianetobessa** [comment]
Hello @spark8899 !

This contribution to the Filecoin codebase was voted as impactful during Q1 and is eligible for a FIL prize as part of a Protocol Labs experiment to reward open source contributions from the community (see[ full details here](https://filecoin.io/blog/posts/a-public-goods-experiment-on-filecoin-retroactively-funding-impact-with-quadratic-voting/)).

Please reach out to us through commons@protocol.ai to receive the instructions on how to claim your reward!

**Pull request merged**